use bank example in chaos test

vivekmenezes opened the pull request:
This is the first of many acceptance tests that will begin to use the SQL API instead of the internal KV one. We should use the KV API only where necessary.

<!-- Reviewable:start -->

[<img src="https://reviewable.io/review_button.svg" height="40" alt="Review on Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/4124)

<!-- Reviewable:end -->

vivekmenezes committed:
use bank example in chaos test

petermattis commented on acceptance/chaos_test.go:
```diff
@@ -56,20 +64,55 @@ func TestChaos(t *testing.T) {
 	// initClient requires that the caller holds the client's write lock.
 	initClient := func(i int) {
-		db, dbStopper := makeClient(t, c.ConnString(i))
+		db := makePGClient(t, c.PGUrl(i))
```
Add a comment that each client is talking to a different node. I almost pointed out that you could share an `sql.DB` across all clients before realizing this was being done.

Oh, and you can assign to `clients[i].db` directly here. Not much point in the `db` variable.

petermattis commented on acceptance/chaos_test.go:
```diff
@@ -225,6 +269,16 @@ func TestChaos(t *testing.T) {
 	}
 
+	// Verify accounts.
+	db = makePGClient(t, c.PGUrl(0))
```
Is there a reason you're create another `db` here? If so, it isn't obvious and should be commented.

petermattis commented:
LGTM

bdarnell commented on acceptance/chaos_test.go:
```diff
@@ -19,35 +19,43 @@
 
 	num := c.NumNodes()
+	if num <= 0 {
+		log.Fatal("%d nodes in cluster", num)
```
Should be `Fatalf`. I wonder why `go vet` didn't catch this?

tamird replied:
because it should also be `t.Fatalf`. `vet` doesn't validate our `log` package.

bdarnell replied:
`vet` doesn't care about packages, it only looks at [function names](https://github.com/golang/tools/blob/a6e3cc80e87d58d7fdb23da870845b456969b929/cmd/vet/print.go#L60-L81), so it should complain about this. Maybe it doesn't look at files that are excluded by a build tag.

However, `Info`/`Infof` isn't on its list, so it doesn't cover most of our use of the `log` package. If we can, we should add `-printfuncs` to our `vet` invocation (although last time I looked at this there were some complexities related to the difference between `go vet` and `go tool vet`. Maybe we've fixed that with the use of `-shadow` now.

tamird replied:
TIL. We should do that - whatever complexities exist wrt `go vet`/`go tool vet` are already handled in the makefile

bdarnell commented on acceptance/chaos_test.go:
```diff
@@ -56,20 +64,55 @@ func TestChaos(t *testing.T) {
+	}
+
+	schema := `
+CREATE TABLE IF NOT EXISTS bank.accounts (
```
It's better to use `DROP TABLE IF EXISTS bank.accounts` and then create it instead of `CREATE TABLE IF NOT EXISTS` followed by truncate. This way you know you're using the right schema instead of a different table that may have been left by a different test run.

vivekmenezes commented:
Review status: 0 of 2 files reviewed at latest revision, 4 unresolved discussions.

---

*[acceptance/chaos_test.go, line 54 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4124#-K9eUI_UVeBeLfrGZIb9-r1-54:-K9hdqZNX71T5HVwmTtY:1385797678) ([raw file](https://github.com/cockroachdb/cockroach/blob/1d8fb7efbf64d1171215d2435af4a7917c721720/acceptance/chaos_test.go#L54)):*
go vet doesn't find this kind of error. changed it to t.Fatalf

---

*[acceptance/chaos_test.go, line 74 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4124#-K9eUI_UVeBeLfrGZIb9-r1-74:-K9heufPdhN8LkXiIPlk:-467915377) ([raw file](https://github.com/cockroachdb/cockroach/blob/1d8fb7efbf64d1171215d2435af4a7917c721720/acceptance/chaos_test.go#L74)):*
fixed

---

*[acceptance/chaos_test.go, line 90 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4124#-K9eUI_UVeBeLfrGZIb9-r1-90:-K9hieGGBjhG10ghp5fJ:-467991002) ([raw file](https://github.com/cockroachdb/cockroach/blob/1d8fb7efbf64d1171215d2435af4a7917c721720/acceptance/chaos_test.go#L90)):*
Good point. Done

---

*[acceptance/chaos_test.go, line 273 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4124#-K9eUI_UVeBeLfrGZIb9-r1-273:-K9hiUIGwEi3PXOJaTZb:-768885343) ([raw file](https://github.com/cockroachdb/cockroach/blob/1d8fb7efbf64d1171215d2435af4a7917c721720/acceptance/chaos_test.go#L273)):*
I've added a comment explaining that we do this because node 0 might have been restarted by the chaos monkey.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4124)*
<!-- Sent from Reviewable.io -->

tamird commented on acceptance/chaos_test.go:
```diff
@@ -225,6 +273,19 @@ func TestChaos(t *testing.T) {
+	db = makePGClient(t, c.PGUrl(0))
+	var sum int
+	if err := db.QueryRow("SELECT SUM(balance) FROM bank.accounts").Scan(&sum); err != nil {
+		log.Fatal(err)
```
this should be `t.Fatal`.

tamird commented:
Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[acceptance/chaos_test.go, line 273 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4124#-K9eUI_UVeBeLfrGZIb9-r1-273:-K9hrrDCVBWe-xxoIs5t:-1689444096) ([raw file](https://github.com/cockroachdb/cockroach/blob/1d8fb7efbf64d1171215d2435af4a7917c721720/acceptance/chaos_test.go#L273)):*
Doesn't `sql.DB` reconnect automatically?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4124)*
<!-- Sent from Reviewable.io -->

vivekmenezes commented:
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[acceptance/chaos_test.go, line 273 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4124#-K9eUI_UVeBeLfrGZIb9-r1-273:-K9hvSQNAS186-xdlhKV:774781026) ([raw file](https://github.com/cockroachdb/cockroach/blob/1d8fb7efbf64d1171215d2435af4a7917c721720/acceptance/chaos_test.go#L273)):*
We see an error  "getsockopt: connection refused"

---

*[acceptance/chaos_test.go, line 283 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4124#-K9eUI_UVeBeLfrGZIb9-r2-283:-K9hvjNTFIGznzaAfuG1:-149754998) ([raw file](https://github.com/cockroachdb/cockroach/blob/002402c4a147c87a0895c676e0a93492f321f191/acceptance/chaos_test.go#L283)):*
thanks. Will send you another PR

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4124)*
<!-- Sent from Reviewable.io -->
